### PR TITLE
Set number of token types to 0 for NMT

### PR DESCRIPTION
**Pull request by MaximumEntropy**
This PR sets the number of token types to 0 by default for NMT since it wasn't being used. Parameters, however, were still being created for it in the model, making it impossible to set`find_unused_parameters=False` in DDP.

### CODE REVIEW

File: nemo/collections/nlp/modules/common/transformer/transformer_utils.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def get_nemo_transformer(
             hidden_act=cfg.get('hidden_act', 'relu'),
             mask_future=cfg.get('mask_future', False),
             pre_ln=cfg.get('pre_ln', False),
+            num_token_types=cfg.get('num_token_types', 2)
```
Comment (ericharper) [inline nemo/collections/nlp/modules/common/transformer/transformer_utils.py:81]: Should the default be 2? Or 0?

File: nemo/collections/nlp/modules/common/transformer/transformer_utils.py [hunk 2]
```diff
@@ -94,6 +95,7 @@ def get_nemo_transformer(
             attn_layer_dropout=cfg.get('attn_layer_dropout', 0.0),
             hidden_act=cfg.get('hidden_act', 'relu'),
             pre_ln=cfg.get('pre_ln', False),
+            num_token_types=cfg.get('num_token_types', 2)
```
Comment (ericharper) [inline nemo/collections/nlp/modules/common/transformer/transformer_utils.py:98]: Should the default be 2? Or 0?

File: nemo/collections/nlp/modules/common/transformer/transformer_utils.py (hunk 1 above)
Comment (MaximumEntropy) [reply nemo/collections/nlp/modules/common/transformer/transformer_utils.py:81]: Not sure what the right default should be. AFAIK, there are maybe 3 different models that use this - NMT, TransformerXL LM, and BERT. NMT and TransformerXL need 0 but BERT needs 2. Neither BERT nor TransformerXL set `num_token_types` in their yaml configs. So I figured it's best to leave this at 2. The other option is to set this to 0 and explicitly set 2 for BERT.

Comment (ericharper) [reply nemo/collections/nlp/modules/common/transformer/transformer_utils.py]: Okay, that's sounds good to me.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Review by okuchaiev — APPROVED** [review]

**Pull request merged**